File: analytics/src/test/java/io/appmetrica/analytics/impl/utils/executors/ClientExecutorFactoryTest.java
package io.appmetrica.analytics.impl.utils.executors;

import io.appmetrica.analytics.testutils.CommonTest;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(RobolectricTestRunner.class)
public class ClientExecutorFactoryTest extends CommonTest {

    private ClientExecutorFactory mClientExecutorFactory;

    @Before
    public void setUp() throws Exception {
        mClientExecutorFactory = new ClientExecutorFactory();
    }

    @Test
    public void testCreateApiProxyExecutor() {
        assertThat(mClientExecutorFactory.createApiProxyExecutor().getLooper().getThread().getName())
                .startsWith("IAA-CAPT");
    }

    @Test
    public void testCreateReportsSenderExecutor() {
        assertThat(mClientExecutorFactory.createReportsSenderExecutor().getLooper().getThread().getName())
                .startsWith("IAA-CRS");
    }

    @Test
    public void testCreateDefaultExecutor() {
        assertThat(mClientExecutorFactory.createDefaultExecutor().getLooper().getThread().getName())
                .startsWith("IAA-CDE");
    }
}

File: analytics/src/test/java/io/appmetrica/analytics/impl/utils/executors/ClientExecutorFactoryTest.kt
package io.appmetrica.analytics.impl.utils.executors

import io.appmetrica.analytics.testutils.CommonTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.kotlin.mock
import org.mockito.kotlin.verify
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class ClientExecutorFactoryTest : CommonTest() {

    private val clientExecutorFactory: ClientExecutorFactory by setUp { ClientExecutorFactory() }

    @Test
    fun createReportsSenderExecutor() {
        assertThat(clientExecutorFactory.createReportsSenderExecutor().looper.thread.name).startsWith("IAA-CRS")
    }

    @Test
    fun createDefaultExecutor() {
        assertThat(clientExecutorFactory.createDefaultExecutor().looper.thread.name).startsWith("IAA-CDE")
    }

    @Test
    fun createInitCoreThread() {
        val runnable = mock<Runnable>()
        val thread = clientExecutorFactory.createInitCoreThread(runnable)
        assertThat(thread.name).startsWith("IAA-INIT_CORE")
        thread.start()
        thread.join()
        verify(runnable).run()
    }
}
